sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BeamSqlUnparseContext.java

```diff
@@ -110,6 +111,26 @@ public SqlNode toSql(RexProgram program, RexNode rex) {
     public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
       writer.literal("DATETIME '" + timestampString.toString() + "'");
     }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      if (!super.equals(o)) {
```

Comment:
This check is a bit confusing. Why use super's equal to check an object of current class?

Something like `super.equals((super) o)` makes more sense (just pseudo code)

Reply:
Ah so this comes from #12348?

Reply:
This was just automatically generated by IntelliJ